src/sentry/services/hybrid_cloud/organization/service.py

```diff
@@ -274,5 +279,40 @@ def update_option(self, *, organization_id: int, key: str, value: OptionValue) -
     def delete_option(self, *, organization_id: int, key: str) -> None:
         pass
 
+    @regional_rpc_method(resolve=ByOrganizationId())
+    @abstractmethod
+    def send_signal(
+        self, *, organization_id: int, signal: RpcOrganizationSignal, args: Mapping[str, int | None]
+    ) -> None:
+        pass
+
+    def schedule_signal(
+        self,
+        signal: Signal,
+        organization_id: int,
+        args: Mapping[str, int | None],
+    ) -> None:
+        from sentry.models.outbox import ControlOutbox, OutboxCategory, OutboxScope, outbox_context
+
+        if SiloMode.get_current_mode() == SiloMode.REGION:
```

Comment:
I think (and you probably agree) that we should avoid looking at the silo mode inside business logic wherever possible, and this is in a gray area as to how much it counts as business logic. I'm fine with this, but if we replicate the pattern into other services ("rule of three" and all), I'd like to try to abstract it out (similar to how `RpcService` abstracts out the mode-sensitive logic for vanilla RPCs).

Reply:
I, agree.  I have an idea, let me get it in.

Reply:
https://github.com/getsentry/sentry/pull/51484/commits/d8f347a1c100fc110970cf68b06322a2d76db4f8